Sources/AuthFoundation/JWT/Internal/DefaultIDTokenValidator.swift

```diff
@@ -18,61 +18,72 @@ import CommonCrypto
 
 struct DefaultIDTokenValidator: IDTokenValidator {
     var issuedAtGraceInterval: TimeInterval = 300
+    var checks: [ValidationCheck] = ValidationCheck.allCases
+    
+    enum ValidationCheck: CaseIterable {
+        case issuer, audience, scheme, algorithm, expirationTime, issuedAtTime, nonce, maxAge, subject
+    }
     
     func validate(token: JWT, issuer: URL, clientId: String, context: IDTokenValidatorContext?) throws {
         guard let tokenIssuerString = token.issuer,
-              let tokenIssuer = URL(string: tokenIssuerString)
-        else {
-            throw JWTError.invalidIssuer
-        }
-
-        guard tokenIssuer.absoluteString == issuer.absoluteString
+              let tokenIssuer = URL(string: tokenIssuerString),
+              tokenIssuer.absoluteString == issuer.absoluteString
         else {
             throw JWTError.invalidIssuer
         }
-
-        guard token[.audience] == clientId else {
-            throw JWTError.invalidAudience
-        }
-
-        guard tokenIssuer.scheme == "https" else {
-            throw JWTError.issuerRequiresHTTPS
-        }
-        
-        guard token.header.algorithm == .rs256 else {
-            throw JWTError.unsupportedAlgorithm(token.header.algorithm)
-        }
         
-        guard let expirationTime = token.expirationTime,
-              expirationTime > Date.nowCoordinated else {
-            throw JWTError.expired
-        }
-        
-        guard let issuedAt = token.issuedAt,
-              abs(issuedAt.timeIntervalSince(Date.nowCoordinated)) <= issuedAtGraceInterval
-        else {
-            throw JWTError.issuedAtTimeExceedsGraceInterval
-        }
-        
-        guard token["nonce"] == context?.nonce else {
-            throw JWTError.nonceMismatch
-        }
-
-        if let maxAge = context?.maxAge {
-            guard let authTime = token.authTime else {
-                throw JWTError.invalidAuthenticationTime
-            }
-            
-            let elapsedTime = issuedAt.timeIntervalSince(authTime)
-            guard elapsedTime > 0 && elapsedTime <= maxAge else {
-                throw JWTError.exceedsMaxAge
+        for check in checks {
+            switch check {
+            case .issuer:
+                guard tokenIssuer.absoluteString == issuer.absoluteString
+                else {
```

Comment:
You can choose to put this `else` on the same line as `guard` if you want. But, with our style, we optimize for readability over purity.